src/dswx_sar/region_growing.py

```diff
@@ -0,0 +1,308 @@
+                                 initial_seed,
+                                 tolerance,
+                                 maxiter):
+
+    """Process region growing for blocks
+
+    Parameters
+    ----------
+    block_param: BlockParam
+        Object specifying where and how much to read and write to out_raster
+    loopind: integer
+        index value for iteration
+    base_dir: str
+        path for fuzzy geotiff and input geotiff
+    fuzzy_base_name: str
+        prefix for temporary file name
+    input_tif_path: str
+        path of initial fuzzy later
+    initial_seed : float
+        Threshold used for the seed of water body [0 ~ 1]
+    tolerance : float
+        relaxed threshold to be used for transient area
+        between water and non-water
+    maxiter : integer
+        maximum iteration for region growing
+
+    Returns
+    ----------
+    block_param: BlockParam
+        Object specifying where and how much to read and write to out_raster
+    data_block: numpy.ndarraya
+        fuzzy values after region growing
+    """
+
+    # At first loop, read block from intial fuzzy value geotiff
+    # Otherwise, read block from previous loop
+    if loopind == 0:
+        fuzzy_map_temp = input_tif_path
+    else:
+        fuzzy_map_temp = \
+            f'{base_dir}/{fuzzy_base_name}_temp_loop_{loopind}.tif'
+    data_block = dswx_sar_util.get_raster_block(
+        fuzzy_map_temp, block_param)
+
+    # Run region growing for fuzzy values
+    region_grow_sub = region_growing(data_block,
+                                     initial_seed=initial_seed,
+                                     tolerance=tolerance,
+                                     maxiter=maxiter)
+    # replace fuzzy values with 1 for the pixels included by region growing
+    data_block[region_grow_sub == 1] = 1
+
+    return block_param, data_block
+
+
+def run_parallel_region_growing(input_tif_path,
+                                output_tif_path,
+                                lines_per_block=200,
+                                initial_seed=0.6,
+                                tolerance=0.45,
+                                maxiter=200):
+    """Process region growing using parallel
+
+    Parameters
+    ----------
+    input_tif_path: str
+        path of fuzzy-logic value Geotiff
+    output_tif_path: str
+        path of region-growing
+    lines_per_block: int
+        lines per block
+    initial_seed: float
+        initial seed values where region-growing starts
+    tolerance: float
+        value where region-growing stops
+    maxiter: integer
+        maximum number of dilation
+    """
+
+    meta_dict = dswx_sar_util.get_meta_from_tif(input_tif_path)
+    data_length = meta_dict['length']
+    data_width = meta_dict['width']
+    data_shape = [data_length, data_width]
+
+    # Process fast region-growing with blocks
+    # In each iteration, the block size will increase to cover 
+    # more areas
```

Comment:
```suggestion
    # more areas to accelerate processing in challenging
    # areas after the initial iteration
```
My attempt to incorporate additional details from explanation below